utils/firestore.js

```
@@ -1410,137 +1410,72 @@ const getSpecimenCollections = async (token, siteCode) => {
-            }
-      }
-      else{
-          if(startDate !== "0"){
-              snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-          }
-          else{
-              snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).orderBy('656548982', 'desc').get();
-          }
-      }
-    }
-    else{
-        if(endDate !== "0"){
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '<=', endDate).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '<=', endDate).orderBy('656548982', 'desc').get();
-            }
-        }
-        else{
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).orderBy('656548982', 'desc').get();
-            }
-        }
+        query = buildQueryWithFilters(query, trackingId, endDate, startDate, source, siteCode)
+
+        const snapshot = await query.orderBy('656548982', 'desc').get();
+        const result = snapshot.docs.length;
+        
+        return result;
+    } catch (error) {
+        console.error(error);
+        return new Error(error)
```

This returned error is not handled by calling function. Better return `[]`.